webapp/src/main/java/org/apache/atlas/web/rest/TypesREST.java

```diff
@@ -483,8 +483,17 @@ public AtlasTypesDef createAtlasTypeDefs(@Context HttpServletRequest servletRequ
         }
     }
 
-    private void validateTypeCreateOrUpdate(AtlasTypesDef typesDef) throws AtlasBaseException {
+    private void validateTypeBeforeAction(AtlasTypesDef typesDef) throws AtlasBaseException {
+        try {
+            validateBuiltInTypeNames(typesDef);
```

Comment:
The issue I saw was related to BM attributes - they were created but weren't available for deletion. This change is effective for all typedefs except BMs. Also, validateBuiltInTypeNames is there to check whether we are passing any built-in type name or not, so the `Typename is invalid` error is thrown from somewhere else